logstash-core/lib/logstash/patches/resolv.rb

```diff
@@ -0,0 +1,36 @@
+require "resolv"
+
+# ref: https://github.com/logstash-plugins/logstash-filter-dns/issues/40
+#
+# JRuby 9k versions prior to 9.1.16.0 have a bug which crashes IP address
+# resolution after 64k unique IP addresses resolutions.
+#
+# Note that the oldest JRuby version in LS 6 is 9.1.13.0 and
+# JRuby 1.7.25 and 1.7.27 (the 2 versions used across LS 5) are not affected by this bug.
+
+# make sure we abort if a known correct JRuby version is installed 
+# to avoid having an unnecessary legacy patch being applied in the future.
+raise("Unnecessary patch on resolv.rb for JRuby version 9.1.16+") if Gem::Version.new(JRUBY_VERSION) >= Gem::Version.new("9.1.16.0")
```

Comment:
Why not just skip the patch like we do in https://github.com/logstash-plugins/logstash-filter-dns/pull/45/files?

Reply:
the idea is that if we hit that condition and raise, tests will break and will be an indication that we can remove that patch altogether. I don't see the point of keeping that patch code around if it is not required. I rather have the tests fail than forget about the useless patch code in the future.